### PR TITLE
EDL improvements

### PR DESCRIPTION
**Pull request by moishce**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/44748
https://github.com/demisto/etc/issues/43330
https://github.com/demisto/etc/issues/40155
https://github.com/demisto/etc/issues/37282

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 1]
```diff
@@ -273,49 +593,56 @@ def is_valid_cidr(cidr: str) -> bool:
...
-                # if port was in the indicator and url_port_stripping param not set - ignore the indicator
-                continue
-            # Reformatting to PAN-OS URL format
-            with_invalid_tokens_indicator = indicator
-            # mix of text and wildcard in domain field handling
-            indicator = _INVALID_TOKEN_REMOVAL.sub('*', indicator)
-            # check if the indicator held invalid tokens
+
+            indicator = url_handler(indicator, request_args.url_protocol_stripping,
+                                    request_args.url_port_stripping, request_args.url_truncate)
```
Comment (ShahafBenYakir) [inline Packs/EDL/Integrations/EDL/EDL.py:638]: Wasn't the `url_handler` already done as part of the `get_indicators_to_format` function?

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 2]
```diff
@@ -273,49 +593,56 @@ def is_valid_cidr(cidr: str) -> bool:
...
+
             if request_args.drop_invalids:
-                if with_invalid_tokens_indicator != indicator:
-                    # invalid tokens in indicator - ignore the indicator
+                if indicator != _PORT_REMOVAL.sub(_URL_WITHOUT_PORT, indicator) or\
+                        indicator != _INVALID_TOKEN_REMOVAL.sub('*', indicator):
+                    # check if the indicator held invalid tokens or port
```
Comment (ShahafBenYakir) [inline Packs/EDL/Integrations/EDL/EDL.py:643]: what are we checking here? can you give an example?

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 3]
```diff
@@ -273,49 +593,56 @@ def is_valid_cidr(cidr: str) -> bool:
...
+        4) if url_truncate, truncate urls
     * Other indicator types:
         1) if drop_invalids, drop invalids (has invalid chars)
         2) if port_stripping, strip ports
     """
-    formatted_indicators = set()
     ipv4_formatted_indicators = set()
     ipv6_formatted_indicators = set()
-    for ioc in iocs:
+    iocs.seek(0)
```
Comment (ShahafBenYakir) [inline Packs/EDL/Integrations/EDL/EDL.py:625]: Where do you close this file?

File: Packs/EDL/Integrations/EDL/EDL_test.py [hunk 4]
```diff
@@ -137,92 +157,77 @@ def test_get_params_port_3(self):
...
+            returned_output = ''
+            not_first_call = False
+            for ioc in iocs_json:
+                returned_output += (create_json_out_format([], ioc, request_args, not_first_call))
+                not_first_call = True
+            returned_output += ']'
+            assert json.loads(returned_output) == iocs_json
 
-        with open('EDL_test/TestHelperFunctions/simplified_demisto_ip_iocs.json', 'r') as iocs_json_f:
+    def test_create_csv_out_format(self):
```
Comment (ShahafBenYakir) [inline Packs/EDL/Integrations/EDL/EDL_test.py:199]: I know you did not write this, but can we add some docstrings here?

**Review by ShahafBenYakir — COMMENTED** [review]
Nice one!
1. See comments in README and apply to the description.md as well.
2. Where are the benchmark numbers we said we are going to mention in the docs?
3. It looks like the unit testing doesn't check many interesting scenarios, for example: other output formats. Do you think you can add such to test for other formats?

File: Packs/EDL/Integrations/EDL/EDL.yml [hunk 5]
```diff
@@ -87,97 +112,178 @@ configuration:
...
   hidden: false
   name: use_legacy_query
   required: false
   type: 8
-description: This integration provides External Dynamic List (EDL) as a service for
-  the system indicators (Outbound feed).
-display: Palo Alto Networks PAN-OS EDL Service
+description: Use the Generic Export Indicators Service integration to provide an endpoint
+  with a list of indicators as a service for the system indicators.
+display: Generic Export Indicators service
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/EDL.yml:195]: Service (capital S)

File: Packs/EDL/Integrations/EDL/EDL_description.md [hunk 6]
```diff
@@ -18,23 +22,35 @@ To access the EDL service by instance name, make sure ***Instance execute extern
...
     * 1 - Collapse to ranges.
     * 2 - Collapse to CIDRs.
- * Example: https://{cortex-xsoar_instance}/instance/execute/{EDL_instance_name}?q="type:ip and sourceBrand:my_source"&tr=1
+ * Example: https://{cortex-xsoar_instance}/instance/execute/{ExportIndicators_instance_name}?q="type:ip and sourceBrand:my_source"&tr=1
+12. **tx** - Whether to output `csv` formats as textual web pages.
+ * Example: https://{cortex-xsoar_instance}/instance/execute/{ExportIndicators_instance_name}?v=csv&tx
 
 ### When running in On-Demand mode
-Make sure you run the `!edl-update` command for the first time to initialize the export process.
+Make sure you run the `!edl-update` command for the first time to initialize the export process.
```
Comment (ytauman) [inline Packs/EDL/Integrations/EDL/EDL_description.md:56]: we should also change this command name

**Review by ShahafBenYakir — COMMENTED** [review]
@moishce few leftovers:
1. Need to modify the EDL pack name to `Generic Export Indicators service`.
2. Change the description to this:
```
Ongoing distribution of indicators from XSOAR to other products in the network, using an endpoint with a list of indicators that can be pulled by external vendors.
```
3. Need to deprecate the following:
   - old export indicators pack 
   - old export indicators integration
   - PAN-OS EDL Service Configuration

File: Packs/EDL/Integrations/EDL/EDL_description.md (hunk 6 above)
Comment (ShahafBenYakir) [reply Packs/EDL/Integrations/EDL/EDL_description.md:56]: **Deprecate this command and add a copy of it with the name: `export-indicators-list-update`

**Review by ShahafBenYakir — APPROVED** [review]
Looks good. see my last comments

File: Packs/EDL/Integrations/EDL/README.md [hunk 7]
```diff
@@ -1,16 +1,4 @@
...
-4. Edit the EDL object on the PAN-OS device to pull from the `PAN-OS EDL Service` instance, as explained in [Access the EDL Service by Instance Name (HTTPS)](#access-the-edl-service-by-instance-name-(https)). You can edit the EDL object using the [panorama-edit-edl](https://xsoar.pan.dev/docs/reference/integrations/panorama#panorama-edit-edl) command in the `Palo Alto Networks PAN-OS` integration.
-5. Commit and push the configuration from the Panorama device to its respective Firewalls using the [PAN-OS Commit Configuration](https://xsoar.pan.dev/docs/reference/playbooks/pan-os-commit-configuration) playbook.
-6. If you have a deployment with 100 firewalls or more, we recommend using your Panorama device and creating an EDL object there, which will be populated from the `PAN-OS EDL Service`. Then push the EDL object to the respective firewalls.
-7. Follow the instructions in the rest of this guide to make sure that the PAN-OS device is connected to the EDL service.
```
[the remark this answers is not in the document]
Comment (moishce) [reply Packs/EDL/Integrations/EDL/README.md:13]: I deleted it by mistake

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 8]
```diff
@@ -336,8 +336,8 @@ def create_json_out_format(list_fields: List, indicator: Dict, request_args: Req
     Returns:
         An indicator to add to the file in json format.
     """
-    if indicator_value := indicator.get('value') and indicator.get('indicator_type') == 'URL':
-        indicator['value'] = url_handler(indicator_value, request_args.url_protocol_stripping,
+    if indicator.get('value') and indicator.get('indicator_type') == 'URL':
```
[the remark this answers is not in the document]
Comment (moishce) [reply Packs/EDL/Integrations/EDL/EDL.py]: revert beak

**Review by ShahafBenYakir — APPROVED** [review]

**Comment by ShahafBenYakir** [comment]
Validation errors are false positives (@moishce pleas open an issue).

**Pull request merged**